### PR TITLE
chore(core): Refactor session 

### PR DESCRIPTION
**Pull request by vrockz747**

### CODE REVIEW

File: common/core/core_session.c [hunk 1]
```diff
@@ -233,165 +172,98 @@ static bool derive_server_public_key() {
...
+                    session_ctx->device.random_priv_key,
+                    session_ctx->server.request_pointer
+                        ->session_random_public,    // TODO: Update proto name
+                                                    // to server_random_pub_key
                     session.session_key) != 0) {
     LOG_ERROR("ERROR: Session key not generated");
     return false;
   }
+  // indicate valid session has been established
+  session.valid = true;
```
Comment (TejasvOnly) [inline common/core/core_session.c:205]: This should be done in `start_request` function

File: common/core/core_session.c [hunk 2]
```diff
@@ -233,165 +172,98 @@ static bool derive_server_public_key() {
...
-    return false;
-  }
-
-  if (!derive_session_iv_and_session_key()) {
-    LOG_ERROR("\nERROR: Generation session keys");
-    return false;
-  }
-
+  // sign payload
+  uint8_t hash[32] = {0};    // hash size reference taken from the atecc_sign
```
Comment (TejasvOnly) [inline common/core/core_session.c:249]:
```suggestion
  uint8_t hash[SHA256_DIGEST_LENGTH] = {0};
```

File: common/core/core_session.c [hunk 3]
```diff
@@ -401,38 +273,65 @@ static void initiate_request(void) {
...
+             session_ctx->server.request_pointer->device_id,
+             DEVICE_SERIAL_SIZE) != 0) {
+    return false;
+  }
+  // derive verification pub key
+  if (!derive_server_public_key()) {
+    LOG_ERROR("\nERROR: Server Randoms not read");
+    return false;
+  }
+  uint8_t hash[32] = {0};
```
Comment (TejasvOnly) [inline common/core/core_session.c:307]:
```suggestion
  uint8_t hash[SHA256_DIGEST_LENGTH] = {0};
```

File: common/core/core_session.c [hunk 4]
```diff
@@ -441,11 +340,15 @@ static void start_request(const core_msg_t *core_msg) {
...
 
 void core_session_clear_metadata() {
-  memzero(&session, sizeof(session_config_t));
+  memzero(&session, sizeof(session_private_t));
 }
 
 void core_session_parse_start_message(const core_msg_t *core_msg) {
   size_t request_type = core_msg->session_start.request.which_request;
+  // used cy_malloc() due to branched flow
+  session_ctx = (session_ctx_t *)cy_malloc(sizeof(session_ctx_t));
```
Comment (TejasvOnly) [inline common/core/core_session.c:349]: I don't think we need `cy_malloc`, we can simple use `malloc` as the `session_ctx` should only be in scope for this function and there are no early exits.

**Review by TejasvOnly — CHANGES_REQUESTED** [review]
Can we also take care of `USE_SIMULATOR` macros in this pr itself?

File: common/core/core_session.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (TejasvOnly) [reply common/core/core_session.c:205]: Not here, but we can simple use `malloc` and `free` in the `core_session_parse_start_message` function.

**Comment by vrockz747** [comment]
> Can we also take care of `USE_SIMULATOR` macros in this pr itself?

I was thinking of fixing it in the error handling PR, but will do in this only

**Comment by vrockz747** [comment]
> Can we also take care of `USE_SIMULATOR` macros in this pr itself?

I was thinking of fixing it in the error handling PR, but will do in this only

File: common/core/core_session.c [hunk 5]
```diff
@@ -215,224 +158,164 @@ static void start_request(const core_msg_t *core_msg);
...
+  ecdsa_get_public_key33(&secp256k1,
+                         session_ctx->device.random_priv_key,
+                         session_ctx->device.random_pub_key);
+  uint32_t offset = 0;
+  // append device_pub_key
+  memcpy(payload + offset,
+         session_ctx->device.random_pub_key,
+         SESSION_PUB_KEY_SIZE);
+  offset += SESSION_PUB_KEY_SIZE;
+  // TODO: standardize simulator handling for hardware specific functionality
```
Comment (TejasvOnly) [inline common/core/core_session.c:212]: Should've been moved with the `#if`

**Review by TejasvOnly — CHANGES_REQUESTED** [review]

File: common/core/core_session.h [hunk 6]
```diff
@@ -51,29 +50,38 @@ typedef enum {
   SESSION_DECRYPT_PACKET_ERR
 } session_error_type_e;
 
-/**
- * @brief Stores the session information
- */
+typedef enum {
+  SESSION_TERMINATED = 0,
+  SESSION_IN_PROGRESS,
+  SESSION_ONGOING
```
[the remark this answers is not in the document]
Comment (vrockz747) [reply common/core/core_session.h:56]: had already updated them in #563

**Pull request merged**